src/main/java/org/apache/solr/benchmarks/BenchmarksMain.java

```diff
@@ -105,32 +108,124 @@ public static void runQueryBenchmarks(List<QueryBenchmark> queryBenchmarks, Stri
+		        	log.info("Loaded " + validations.size() + " validations from " + benchmark.validationFile);
+		        	
+			        for (SolrBenchQuery sbq: generatedQueries) {
+			        	int numFound = getNumFoundFromSolrQueryResponse(sbq.response);
+			        	Map<String, Object> facets = getFacetsFromSolrQueryResponse(sbq.response);
+			        	String key = sbq.queryString.replace('\n', ' ').replace('\r', ' ');
+			        	int expectedNumFound = validations.get(key).first();
+			        	String facetsString = new ObjectMapper().writeValueAsString(facets).replace('\n', ' ');
+			        	String expectedFacetsString = validations.get(key).second();
+			        	if (numFound == expectedNumFound && facetsString.trim().equals(expectedFacetsString.trim())) {
+			        		success++;
+			        	} else {
+			        		log.error("Validation failed: numFound=" + numFound+", expected=" + expectedNumFound + ", facets=" + facetsString +", expected="+expectedFacetsString);
+			        		failure++;
+			        	}
+			        }
+			        
+			        log.info("Successes: "+success+", failures: "+failure);
+
+		        }
+
 		        long time = System.currentTimeMillis() - start;
 		        System.out.println("Took time: " + time);
 		        if (time > 0) {
-		            System.out.println("Thread: " + threads + ", Median latency: " + controlledExecutor.stats.getPercentile(50) +
-		                    ", 95th latency: " + controlledExecutor.stats.getPercentile(95));
-		            ((List)results.get("query-benchmarks").get(benchmark.name)).add(
-		            		Util.map("threads", threads, "50th", controlledExecutor.stats.getPercentile(50), "90th", controlledExecutor.stats.getPercentile(90), 
-		            				"95th", controlledExecutor.stats.getPercentile(95), "mean", controlledExecutor.stats.getMean(), "total-queries", controlledExecutor.stats.getN(), "total-time", time));
+		            Map<String, Number> taskResults = new LinkedHashMap<>();
+		            taskResults.put("threads", threads);
+		            taskResults.put("50th", controlledExecutor.stats.getPercentile(50));
+		            taskResults.put("90th", controlledExecutor.stats.getPercentile(90));
+		            taskResults.put("95th", controlledExecutor.stats.getPercentile(95));
+		            taskResults.put("mean", controlledExecutor.stats.getMean());
+		            taskResults.put("total-queries", controlledExecutor.stats.getN());
+		            taskResults.put("total-time", time);
+		            if (StressMain.validate) taskResults.put("validations-succeeded", success);
+		            if (StressMain.validate) taskResults.put("validations-failed", failure);
+
+		            System.out.println("Query task results: " + taskResults);
+		            ((List)results.get("query-benchmarks").get(benchmark.name)).add(taskResults);
 		        }
 		    }
 		}
 	}
 
+	private static Map<String, Pair<Integer, String>> loadValidations(QueryBenchmark benchmark) throws IOException {
```

Comment:
As per pervious thought about make the validation logic abstract, we might be able to move all the `loadValidations` (and other logic such as preprocessing logic, string replacement, validation file export/import logic) into the concrete implementation of that. 

This could avoid adding more logic into BenchmarksMain 😊